File: scripts/mysql_prepared.lua
-- scripts/mysql_prepared.lua
--
-- MySQL prepared-statement workload for wrkx (ADR 0005, P6-5).
-- Demonstrates per-request parameterization via mysql.execute().
-- Use with:
--   ./wrkx -t4 -c100 -d10s -R500 -s scripts/mysql_prepared.lua \
--          mysql://wrkx:secret@localhost/wrkx

local stmt = mysql.prepare("SELECT ?")
local counter = 0

function request()
    counter = counter + 1
    return mysql.execute(stmt, tostring(counter % 1000))
end

File: scripts/mysql_prepared.js
// scripts/mysql_prepared.js
//
// MySQL prepared-statement workload for wrkx (ADR 0005, P6-5).
// QuickJS equivalent of mysql_prepared.lua.
// Use with:
//   ./wrkx -t4 -c100 -d10s -R500 -s scripts/mysql_prepared.js \
//          mysql://wrkx:secret@localhost/wrkx

const stmt = mysql.prepare("SELECT ?");
let counter = 0;

function request() {
    counter++;
    return mysql.execute(stmt, String(counter % 1000));
}
